src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,34 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    // this._confirmButton = this._popupElement.querySelector("#confirm-button");
+    //this._submitForm = this._submitForm.bind(this);
+  }
+
+  setSubmitAction(action) {
+    this._handleFormSubmit = action;
+  }
+
+  setLoading(isLoading, submitSave) {
+    if (isLoading) {
+      this._popupElement.querySelector(".modal__button").textContent =
+        "Deleting...";
+    } else {
+      this._popupElement.querySelector(".modal__button").textContent =
+        submitSave;
+    }
+  }
+
+  close() {
+    this._popupForm.removeEventListener("submit", this._handleFormSubmit);
```

Comment:
* When I delete 3 cards in a row there are errors  http://joxi.ru/D2PnJ9QCWXD6Zr


You need to make `_submitForm ` and inside it you'll call `_handleFormSubmit` which you update every time you opent the modal


```js

  _submitForm = () => {
    this._handleFormSubmit();
  };
  close() {
    this._popupForm.removeEventListener("submit", this._submitForm);  // here you set `_submitForm `
    super.close();
  }

  setEventListeners() {
    super.setEventListeners();
    this._popupForm.addEventListener("submit", this._submitForm); // here you remove  `_submitForm `
  }
```


So, this way the function will be correctly updated by `setSubmitAction`

Reply:
I believe this has been corrected.